include/deal.II/hp/dof_level.h

```diff
@@ -296,15 +317,16 @@ namespace internal
       Assert (obj_index < dof_offsets.size(),
               ExcIndexRange (obj_index, 0, dof_offsets.size()));
 
-      // make sure we are on an
-      // object for which DoFs have
-      // been allocated at all
+      // make sure we are on an object for which DoFs have been
+      // allocated at all
       Assert (dof_offsets[obj_index] != (offset_type)(-1),
               ExcMessage ("You are trying to access degree of freedom "
                           "information for an object on which no such "
                           "information is available"));
 
-      Assert (fe_index == active_fe_indices[obj_index],
+      Assert (fe_index == ((signed_active_fe_index_type)active_fe_indices[obj_index]>=0 ?
```

Comment:
These three lines confuse me. What are you doing here? `active_fe_indices` is `unsigned` you cast it to a `signed` and check that it is not negative but this could only happened if the `active_fe_indices[obj_index]` is large, right? Can't you just check the value of `active_fe_indices[obj_index]` directly?